src/Development/IDE/Core/Compile.hs

```diff
@@ -370,7 +370,7 @@ getModSummaryFromBuffer fp contents dflags parsed = do
 -- parsed module (or errors) and any parse warnings. Does not run any preprocessors
 parseFileContents
        :: GhcMonad m
-       => (GHC.ParsedSource -> IdePreprocessedSource)
+       => (GHC.ParsedModule -> IdePreprocessedSource)
```

Comment:
changing the return type seems like a cleaner way. You can mess things up either way, don’t write a broken transformation :slightly_smiling_face: That said, I don’t care too much about this so your choice.